tests/c/loopy_funcs_not_inlined_by_default.c

```diff
@@ -0,0 +1,52 @@
+// Run-time:
+//   env-var: YKD_SERIALISE_COMPILATION=1
+//   env-var: YKD_PRINT_IR=jit-pre-opt
+//   stderr:
+//     ...
+//     --- Begin jit-pre-opt ---
+//     ...
+//     define i8 @__yk_compiled_trace_0(ptr %0, ptr %1, i64 %2, ptr %3) {
+//       ...
+//       call void @never_inline_into_trace(i32 noundef ...
+//       ...
+//     }
+//     ...
+//     --- End jit-pre-opt ---
+
+// Check a loopy function that is NOT annotated `yk_unroll_safe` never gets
+// inlined into the trace.
+
+#include <assert.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+#include <yk.h>
+#include <yk_testing.h>
+
+int call_me(int x); // from extra_linkage/call_me.c
+
+// A function containing a loop, not marked `yk_unroll_safe`, and never inlined
+// during AOT compilation.
+__attribute__((noinline)) void never_inline_into_trace(int x) {
```

Comment:
I had to stare at this for a while to realise why `__attribute__((noinline))` is here. Let's maybe add a comment along the lines of "We need __attribute__((noinline)) so that this function isn't statically inclined by clang, at which point we wouldn't be testing yk's loop unrolling heuristic on this function"?

Reply:
Fixed in [35b29c6](https://github.com/ykjit/yk/pull/564/commits/35b29c65b274d28f77a3faa7d0a29bda80d7e168).